src/callback_requests.rs

```diff
@@ -0,0 +1,23 @@
+use anyhow::Result;
+use tokio::sync::oneshot;
+
+/// Holds the response to a waPC evaluation request
+#[derive(Debug)]
+pub struct CallbackResponse {
+    /// The data to be given back to the waPC guest
+    pub payload: Vec<u8>,
+}
+
+/// Describes the different kinds of request a waPC guest can make to
+/// our host.
+#[derive(Debug)]
+pub enum CallbackRequest {
```

Comment:
What do you think about `CallbackRequest` being parameterized with the response type and having a two-level request, given every enum variant will have the response. Pseudorust:

```rust
pub enum CallbackRequestType {
  OciManifestDigest { image: String },
}

pub struct CallbackRequest<R> {
  request: CallbackRequestType,
  response_channel: oneshot::Sender<Result<R>>,
}
```

Reply:
This is a really good idea. I'll change that

Reply:
Done